File: Programmers/Hash/위장.py
def solution(clothes):
    answer = 1
    hash_ = dict()
    for i in range(len(clothes)):
        try:
            hash_[clothes[i][1]] += 1
        except:
            hash_[clothes[i][1]] = 1
    for i in list(map(lambda x:x+1, hash_.values())):
        answer *= i
    return answer - 1

File: Programmers/Hash/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    map<string, int> m;
    for(auto i = clothes.begin(); i != clothes.end(); ++i){
        if(m[(*i)[1]])
            m[(*i)[1]] += 1;
        else
            m[(*i)[1]] = 1;;
    }
    
    for(auto i = m.begin(); i != m.end(); ++i)
        answer *= i->second + 1;
    
    return answer -1;
}